src/api/03-user-experience/09-google-cast.md

```diff
@@ -0,0 +1,247 @@
+        //    if it fails, be prepared to handle the error.
+        await session.loadMedia(request);
+      } catch (error) { /* Handle the error. */ }
+    }
+  }
+}
+</script>
+<script src="https://www.gstatic.com/cv/js/sender/v1/cast_sender.js?loadCastFramework=1"></script>
+
+<video src="video.mp4"></video>
+
+<!-- Custom element defined by the framework. -->
+<google-cast-launcher></google-cast-launcher>
+```
+
+## Metadata
+
+Receiver applications may expose additional information about the media in their UI, or provide additional features depending on metadata passed to them.
+
+The [Cast Application Framework] defines several [Metadata classes] that developers can use to convey information like the content title, thumbnail, release dates, authors etc.
+
+```js
+const mediaInfo = new chrome.cast.media.MediaInfo(
+  'http://localhost/video.mp4',                // Provide a URL.
+  'video/mp4; codecs="avc1.640032,mp4a.40.2"', // MIME + codecs.
+);
+
+// Image URLs need to be "wrapped" by the Image class.
+const thumbnail = new chrome.cast.Image('http://localhost/thumb.jpg');
+
+// Our video is a short clip, but if it was a full-length movie,
+// we would use `MovieMediaMetadata` here.
+mediaInfo.metadata = new chrome.cast.media.GenericMediaMetadata();
+mediaInfo.metadata.title = 'My Dog Chasing Geese';
+mediaInfo.metadata.images = [ thumbnail ];
+```
+
+## Captions and subtitles
+
+Media can contain additional [tracks of different types]. You can use text tracks to provide captions or subitles for your videos.
+
+```js
+const mediaInfo = new chrome.cast.media.MediaInfo(
+  'http://localhost/video.mp4',                // Provide a URL.
+  'video/mp4; codecs="avc1.640032,mp4a.40.2"', // MIME + codecs.
+);
+
+// Instantiate the Track object first.
+const captionsTrack = new chrome.cast.media.Track(
+  'captions-en', // Unique track identifier.
+  chrome.cast.media.TrackType.TEXT,
+);
+
+captionsTrack.trackContentId = 'http://localhost/captions-en.vtt';
+captionsTrack.subtype = chrome.cast.media.TextTrackType.CAPTIONS;
+captionsTrack.name = 'English CC';
+captionsTrack.language = 'en-US'; // RFC 5646 Language Tag.
+captionsTrack.trackContentType = 'text/vtt';
+
+// The `MediaInfo` object may contain several tracks of various types,
+// how and whether additional tracks are exposed in the UI depends
+// on the receiver application implementation.
+mediaInfo.tracks = [ captionsTrack ];
+```
+
+## UX considerations
+
+When the casting session starts, video playback pauses in the sender application and the `<google-cast-launcher>` changes its styling to indicate the session in progress.
+
+However in many cases it is useful to provide additional indication. One of the common patterns is rendering a video overlay along with displaying a name of the device the media is being casted to.
+
+To achieve this, we can listen for `SESSION_STATE_CHANGED` events and add or remove an overlay element when session is started and ended:
+
+```html
+<style>
```

Comment:
I'm not sure CSS is necessary in this context as we can use html attribute hidden to hide button

Reply:
The `.hidden` class is applied to the video overlay that shows a "Casting to [device name]" message on top of the original video element rect.

Reply:
I was suggesting using `overlay.setAttribute("hidden", "");` and `overlay.removeAttribute("hidden")` in JS code and removes the CSS to make the code sample shorter. WDYT?
I don't feel strongly about it though ;)

Reply:
Did you remove CSS as well?